bindings/python/src/lib.rs

```diff
@@ -300,17 +300,45 @@ fn find_cudart(module: &PyModule) -> Option<Library> {
     if var != "1" {
         return None;
     }
-    let path: std::path::PathBuf = module
+
+    let path: PyResult<std::path::PathBuf> = module
         .getattr(intern!(module.py(), "_C"))
         .ok()?
         .getattr(intern!(module.py(), "__file__"))
         .ok()?
-        .extract()
-        .ok()?;
-    // SAFETY: This is unsafe because the library might run arbitrary code
-    // So it's really important to make sure we are targeting the correct
-    // library.
-    let lib = unsafe { Library::new(path).ok()? };
+        .extract();
+
+    #[cfg(not(target_os = "windows"))]
+    let lib = {
+        let path = path.ok()?;
+        // SAFETY: This is unsafe because the library might run arbitrary code
+        // So it's really important to make sure we are targeting the correct
+        // library.
+        unsafe { Library::new(path).ok()? }
```

Comment:
Is there a safe alternative to this ?

Reply:
Not for Linux. 

https://docs.rs/libloading/latest/libloading/struct.Library.html

What makes it safe is that the library we're loading is already loaded by torch itself. On Windows, there's a way to load only already loaded librairies making the operation safe, but it seems the flag for it on Linux is not widely available:https://github.com/nagisa/rust_libloading/issues/110